portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -11,22 +11,111 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 package com.google.sps.servlets;
-
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
+
+
+
+
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  final int defaultdisplayNumber = 10;
+  final String commentEntityType = "Comment"; 
+  final String statementProperty = "statement";
+  final String timestampProperty = "timestamp";
+  final String userProperty = "user";
+  int displayNumber = defaultdisplayNumber;
+  UserService userService = UserServiceFactory.getUserService();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query(commentEntityType).addSort(timestampProperty, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery  results = datastore.prepare(query);
+    
+    int countComments = 0;
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()){
+      if(displayNumber == countComments){
+        break;
+      }
+      countComments++;
+      String statement = (String)entity.getProperty(statementProperty);
+      long timestamp = (long)entity.getProperty(timestampProperty);
+      String user = (String)entity.getProperty(userProperty);
+      Comment newComment = new Comment(statement, timestamp, user);
+      comments.add(newComment);
+    }
+
+    response.setContentType("application/json;");
+    if(!userService.isUserLoggedIn()){
+      return;
+    }
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    displayNumber = getNumberOfComments(request);
```

Comment:
Shouldn't we get number of comments from Datastore and not from request?

Reply:
the user can provide the number of comments to be displayed which we get from request.

Reply:
Ok... So there are two functionality in this post request (submitting a comment/no. of comments to display). Please add a comment here that this is for displaying number of comments.